### PR TITLE
Remove dialogue checks upon publication

### PR DESCRIPTION
**Pull request by lagerros**
If two users matched via matching and both filled in forms, and then made a dialogue, we uncheck their checkboxes once that dialogue gets published.

In addition, we soft delete their matchmaking forms, so that, if they match again, they get to fill them out anew. 

:white_check_mark: tested and things seemed to work fine
:white_check_mark: went through all mentions in the codebase of "matchPreference" to check for downstream effects, and updated soft delete behavior in all the relevant cases I found

### CODE REVIEW

File: packages/lesswrong/lib/collections/dialogueMatchPreferences/schema.ts [hunk 1]
```diff
@@ -96,7 +96,17 @@ const schema: SchemaType<DbDialogueMatchPreference> = {
     canCreate: ['admins'],
     canRead: ['members', 'admins'],
     canUpdate: ['admins'],
-  }
+  },
+  // deleted: Indicates whether a form has been soft deleted, after a dialogue was published and the form was reset
+  deleted: {
+    type: Boolean,
+    nullable: false,
```
Comment (b0b3rt) [inline packages/lesswrong/lib/collections/dialogueMatchPreferences/schema.ts:103]: With the nullability PRs merged in, `nullable: false` is now provided by default by `schemaDefaultValue`, `arrayOfForeignKeysField`, and `denormalizedCountOfReferences`.  So it doesn't need to be explicitly specified in those cases (unless you want to explicitly make it nullable, in which case you need to put `nullable: true` _after_ `...schemaDefaultValue(...)` or whatever).

**Review by jpaddison3 — COMMENTED** [review]
No objections from the EA Forum

File: packages/lesswrong/server/callbacks/postCallbacks.ts [hunk 2]
```diff
@@ -563,3 +565,34 @@ getCollectionHooks("Posts").updateAfter.add(async (post: DbPost, props: CreateCa
...
+        await updateMutator({ // soft delete topic form
+          collection: DialogueMatchPreferences,
+          documentId: matchForm._id,
+          set: { deleted: true },
+          currentUser: adminContext.currentUser,
+          context: adminContext,
+          validate: false
+        });
+      }
+    }
```
[the remark this answers is not in the document]
Comment (b0b3rt) [reply packages/lesswrong/server/callbacks/postCallbacks.ts:595]: Really the clean thing to do there would be to factor out this into its own function:
```
async function resetDialogueMatch(dialogueMatchPreferences: DialogueMatchPreferences) {
  const dialogueCheck = await DialogueChecks.findOne(matchForm.dialogueCheckId);
  if (dialogueCheck) {
    await Promise.all([
      updateMutator(...) // reset check
      updateMutator(...) // soft delete topic form
    ]);
  }
}
```

then:
```
await Promise.all(matchForms.map(resetDialogueMatch));
```

**Review by b0b3rt — APPROVED** [review]

File: packages/lesswrong/server/callbacks/postCallbacks.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jpaddison3) [reply packages/lesswrong/server/callbacks/postCallbacks.ts:595]: This is a good PR review comment. There are many places in the codebases where turning around mutations to the user takes too long, and I believe (without checking) that it's due to devs not thinking about this exact issue. (Which definitely happens, I just haven't checked if that's the cause.)

**Pull request merged**